hivelocity/instances_v2.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hivelocity
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hivelocity/hivelocity-cloud-controller-manager/client"
+	"github.com/hivelocity/hivelocity-cloud-controller-manager/pkg/hvutils"
+	corev1 "k8s.io/api/core/v1"
+	cloudprovider "k8s.io/cloud-provider"
+)
+
+// HVInstancesV2 implements cloudprovider.InstanceV2.
+type HVInstancesV2 struct {
+	client client.Interface
+}
+
+var _ cloudprovider.InstancesV2 = &HVInstancesV2{}
+
+// newHVInstanceV2 creates a new HVInstancesV2 struct.
+func newHVInstanceV2(c client.Interface) *HVInstancesV2 {
+	return &HVInstancesV2{client: c}
+}
+
+// getHivelocityDeviceIDFromNode returns the deviceID from a Node.
+// Example: If Node.Spec.ProviderID is "hivelocity://123", then 123
+// will be returned.
+func getHivelocityDeviceIDFromNode(node *corev1.Node) (int32, error) {
+	providerPrefix := providerName + "://"
+	if !strings.HasPrefix(node.Spec.ProviderID, providerPrefix) {
+		return 0, fmt.Errorf("node: %s, ProviderID: %q: %w", node.GetName(), node.Spec.ProviderID,
+			errMissingProviderPrefix)
+	}
+	deviceID, err := strconv.ParseInt(strings.TrimPrefix(node.Spec.ProviderID, providerPrefix), 10, 32) //nolint:revive // https://www.reddit.com/r/golang/comments/10mi0el/strconvparseintmyid_10_32_avoid_magic_numbers/
```

Comment:
Add a ParseInt32 function in pkg/parse. This is what my suggestion was last week as well and is also suggested in the Reddit link.